mimic/test/test_fastly.py

```diff
@@ -0,0 +1,240 @@
+from twisted.trial.unittest import SynchronousTestCase
+from twisted.internet.task import Clock
+
+from mimic.canned_responses import fastly
+from mimic.core import MimicCore
+from mimic.resource import MimicRoot
+from mimic.test.helpers import json_request
+
+
+class FastlyAPITests(SynchronousTestCase):
+
+    """
+    Tests for the Fastly api
+    """
+
+    def setUp(self):
+        """
+        Create a :obj:`MimicCore` with :obj:`FastlyApi` as the only plugin,
+        and create a service
+        """
+        self.core = MimicCore(Clock(), [])
+        self.root = MimicRoot(self.core).app.resource()
+        self.uri = '/fastly'
+        self.fastly_response = fastly.FastlyResponse()
+
+        self.customer_id = 42
+        self.service_name = 'yumyum'
+
+        (self.response, self.service_json) = self.successResultOf(json_request(
+            self, self.root, "POST",
+            self.uri + '/service?customer_id={0}&name={1}'.format(
+                self.customer_id, self.service_name)))
+
+        self.service_id = self.service_json['versions'][0]['service_id']
+        self.version_id = self.service_json['versions'][0]['number']
+
+    def test_get_customer(self):
+        """
+        Test to verify : ``GET /current_customer``
```

Comment:
For test docstrings, I really prefer a style where it's a declarative statement about the functionality being tested.  For example,

```
``GET /current_customer`` against the fastly API mock returns JSON-serialized customer details.
```

This says not only that it's verifying `/current_customer`, but *what* about `/current_customer` is being verified.  This is helpful as APIs grow more behaviors over time, because if an endpoint has 5 or 6 different behaviors depending on its state, a lot of them just end up being `verify GET /current_customer when stuff is different`, which makes a maintainer's job difficult because they have to decipher what is expected to happen when stuff is "normal" and when it is "different" and how those differences arise.